tests/yam/agama/agama_cli.pm

```diff
@@ -0,0 +1,69 @@
+## Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Perform a default installation using only CLI while exercising various commands.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base Yam::Agama::agama_base;
+use testapi;
+use power_action_utils qw(power_action);
+use Utils::Architectures qw(is_s390x);
+use Utils::Backends qw(is_svirt);
+
+sub agama_config_edit {
+    my $regex = shift;
+
+    script_run('agama config edit', timeout => 0);
+    wait_still_screen();
+    type_string($regex);
+    type_string(":wq\n");
+    wait_still_screen();
+}
+
+sub run {
+    my $self = shift;
+
+    validate_script_output('agama 2>&1 | tee', qr/Agama's command-line interface/, proceed_on_failure => 1);
+
+    assert_script_run('agama config show | jq -C');
+
+    assert_script_run("jq -n '.root.password = \"$testapi::password\"' | agama config load");
+    assert_script_run("agama config show | grep $testapi::password");
+
+    my $product_id = get_var('AGAMA_PRODUCT_ID');
+    assert_script_run("jq -n '.product.id = \"$product_id\"' | agama config load");
+    assert_script_run("agama config show | grep $product_id");
+
+    validate_script_output('agama probe', qr/Analyze/);
+
+    my $rpm_url = data_url('yam/agama/hello-world-0.1-1.1.noarch.rpm');
+    assert_script_run("agama download $rpm_url /tmp/hello-world.rpm");
+    validate_script_output('stat /tmp/hello-world.rpm', qr/Size: 7019/);
+
+    agama_config_edit(":\%s/bernhard/jose/g\n");
+    assert_script_run('agama config show | grep jose');
+    agama_config_edit(":\%s/jose/bernhard/g\n");
+
+    script_run('agama events > /tmp/agama_events.log 2>&1 &', timeout => 0);
+    validate_script_output('cat /tmp/agama_events.log', qr/ClientConnected/);
+
+    script_run('agama auth login', timeout => 0);
+    type_string("$testapi::password\n");
+    my $regex = 'Not authenticated in localhost';
```

Comment:
This is not a regex, compiled regex are stored in Perl like my `$var = `/pattern/`` or `qr(pattern)`
As you don't use it more than once, avoid to declare the variable and include it below directly.

Reply:
also if the developer change the message we might missed, would be possible to check for the positive result?